sqlglot/expressions.py

```diff
@@ -5282,6 +5282,11 @@ class ArrayToString(Func):
     _sql_names = ["ARRAY_TO_STRING", "ARRAY_JOIN"]
 
 
+# https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#string
+class String(Func):
+    arg_types = {"this": True, "zone": False}
```

Comment:
TODO: check whether this has unintended side effects, i.e. are there dialects that support `STRING(...)`? Spark definitely has a variant where `STRING(x)` is essentially `CAST(x AS STRING)`.

Reply:
I did a pass over the main dialects, it's supported only for the Hive hierarchy as you mentioned. Will add test cases for that

Reply:
Nice, thanks for checking.